gadget/gadget_test.go

```diff
@@ -3852,3 +3876,63 @@ kernel-cmdline:
 		}
 	}
 }
+
+func (s *gadgetYamlTestSuite) testVolumeMinSize(c *C, gadgetYaml []byte, volSizes map[string]quantity.Size) {
+	ginfo, err := gadget.InfoFromGadgetYaml(gadgetYaml, nil)
+	c.Assert(err, IsNil)
+
+	c.Assert(len(ginfo.Volumes), Equals, len(volSizes))
+	for k, v := range ginfo.Volumes {
+		c.Logf("checking size of volume %s", k)
+		c.Check(v.MinSize(), Equals, quantity.Size(volSizes[k]))
+	}
+}
+
+func (s *gadgetYamlTestSuite) TestVolumeMinSize(c *C) {
+	for _, tc := range []struct {
+		gadgetYaml []byte
+		volsSizes  map[string]quantity.Size
+	}{
+		{
+			gadgetYaml: gadgetYamlUnorderedParts,
+			volsSizes: map[string]quantity.Size{
+				"myvol": 1300 * quantity.SizeMiB,
+			},
+		},
+		{
+			gadgetYaml: mockMultiVolumeUC20GadgetYaml,
+			volsSizes: map[string]quantity.Size{
+				"frobinator-image":  (1 + 500 + 10 + 500 + 1024) * quantity.SizeMiB,
+				"u-boot-frobinator": 24576 + 623000,
+			},
+		},
+		{
+			gadgetYaml: mockMultiVolumeGadgetYaml,
+			volsSizes: map[string]quantity.Size{
+				"frobinator-image":  (1 + 128 + 380) * quantity.SizeMiB,
+				"u-boot-frobinator": 24576 + 623000,
+			},
+		},
+		{
+			gadgetYaml: mockVolumeUpdateGadgetYaml,
+			volsSizes: map[string]quantity.Size{
+				"bootloader": 12345 + 88888,
+			},
+		},
+		{
+			gadgetYaml: gadgetYamlPC,
+			volsSizes: map[string]quantity.Size{
+				"pc": (1 + 1 + 50) * quantity.SizeMiB,
+			},
+		},
+		{
+			gadgetYaml: gadgetYamlUC20PC,
+			volsSizes: map[string]quantity.Size{
+				"pc": (1 + 1 + 1200 + 750 + 16 + 1024) * quantity.SizeMiB,
+			},
+		},
+	} {
+		c.Logf("test min size for %s", tc.gadgetYaml)
+		s.testVolumeMinSize(c, tc.gadgetYaml, tc.volsSizes)
```

Comment:
(nitpick) I wonder what the advantage is to extract this as a helper function? It seems just inlining it here would be fine(?). But fine either way :)